package.json

```diff
@@ -113,6 +113,36 @@
           "type": "string",
           "default": "meson",
           "description": "Specify the meson executable to use"
+        },
+        "mesonbuild.muonPath": {
+          "type": "string",
+          "default": "muon",
+          "description": "Specify the muon executable to use"
+        },
+        "mesonbuild.linting.enabled": {
+          "type": "boolean",
+          "default": true,
+          "description": "Globally enable/disable linting"
+        },
+        "mesonbuild.linter.muon.enabled": {
+          "type": "boolean",
+          "default": false,
+          "description": "Enable muon linter"
+        },
+        "mesonbuild.formatting.enabled": {
+          "type": "boolean",
+          "default": true,
+          "description": "Globally enable/disable formatting"
+        },
+        "mesonbuild.formatter.muon.enabled": {
```

Comment:
I don't these need to exist anymore, the `*.muon.enabled` settings. We should just have `mesonbuild.linter` which user can set to `muon` and `mesonbuild.formatter` which user can also set to `muon`. Those should be the defaults.

Reply:
The linting framework was designed to support multiple linters (like [ale](https://github.com/dense-analysis/ale)) simultaneously.

The formatting framework is basically the same as the linting one so it also supports multiple formatters simultaneously, though I'm not sure what vscode does if it has multiple registered formatters.

Reply:
You can pick a default formatter if more than one exists.

Reply:
So given what I said, do you still think the options need to be changed?

Reply:
Alright, I kept the formatting config the same, but I changed the formatting config to have a provider.  I think I like having enabled / provider separate rather than having `"none"` be a provider so I left it that way for now.